apps/project/serializers.py

```diff
@@ -325,6 +325,14 @@ class Meta:
         model = Project
         exclude = ('members', )
 
+    def to_representation(self, data):
+        result = super().to_representation(data)
+        if not result['is_private'] and self.context['request'].user.id in [e['member'] for e in result['memberships']]:
+            pass
+        else:
+            result['memberships'] = []
+        return result
+
```

Comment:
Instead of this, we should create 2 serializers (one for members and one for public view) and only show members for members. This way we can add additional fields for member view later also. You can create a **ProjectSerailizer** for non-member and inheritance class **ProjectMemberViewSerializer** (or better naming?) for members with additional fields.
cc: @annshress 

@AdityaKhatri Is member data used in the client from project API? If not we can remove this entirely for now.